File: frontend/src/locales/backtest.ts
import type { Messages } from './index'

/**
 * backtest —— 策略回测（views/BacktestView.vue + 首页「最近回测」摘要）。
 *
 * metric.* 是**后端 JSON 指标字段名的显示译文**：键名沿用后端中文字段名
 * （总收益率 / 夏普比率 / VaR (95%)…），因为它们是 API 契约的一部分，
 * 视图里 `m.总收益率` 这类访问必须与后端一致；只有右侧译文随语言切换。
 */

export const zh: Messages = {
  subtitle: '选择市场、策略与标的，运行回测查看动态绩效图表（Alpha/Beta、回撤、日收益）',
  running: '正在运行回测，请稍候...',
  /** toast：从首页「重跑」带入历史参数时的提示 */
  formRestored: '已回填历史回测参数：{name} · {code}',
  strategiesLoadFailed: '策略列表加载失败：',
  runFailed: '回测失败：',
  done: '回测完成',
  selectStrategyRequired: '请选择策略',
  noTrades: '回测期间无成交记录（策略未触发买卖信号）',
  /** K 线买入加权平均成本线标注 */
  avgCost: '成本均价',
  market: {
    zh_a: 'A 股',
    us: '美股',
  },
  /** 货币单位词（y 轴名 / 风险面板 VaR 列头）；具体金额一律走 lib/format 的 fmtCurrency */
  currency: {
    cny: '元',
    usd: '美元',
  },
  benchmark: {
    csi300: '沪深300',
    sp500: '标普500',
  },
  stockHint: {
    zh_a: '不带前缀的 6 位 A 股代码',
    us: '美股字母代码，如 AAPL、MSFT、BRK.B',
  },
  commissionHint: {
    zh_a: 'A 股已自动叠加印花税（卖出 0.05%）与过户费（双边 0.001%）',
    us: '美股仅佣金（双边），无印花税/过户费',
  },
  form: {
    market: '市场',
    strategy: '选择策略',
    strategyPlaceholder: '请选择策略',
    stockCode: '股票代码',
    initialCapital: '初始资金',
    commissionRate: '佣金费率',
    startDate: '开始日期',
    endDate: '结束日期',
    run: '运行回测',
  },
  section: {
    metrics: '绩效指标',
    tradesKline: '回测 K 线 · 买卖点标注',
    equity: '累计收益曲线',
    drawdown: '最大回撤曲线',
    dailyReturns: '日收益率分布',
  },
  empty: {
    kline: 'K线数据加载失败或回测区间无效',
    result: '运行回测后在此查看动态绩效图表',
  },
  metric: {
    总收益率: '总收益率',
    年化收益率: '年化收益率',
    夏普比率: '夏普比率',
    最大回撤: '最大回撤',
    胜率: '胜率',
    盈亏比: '盈亏比',
    信息比率: '信息比率',
    年化波动率: '年化波动率',
    卡玛比率: '卡玛比率',
    下行标准差: '下行标准差',
    var95: 'VaR(95%)',
    cvar95: 'CVaR(95%)',
    交易次数: '交易次数',
    盈利交易数: '盈利交易',
    亏损交易数: '亏损交易',
  },
  series: {
    strategyNav: '策略净值',
    drawdown: '回撤',
    dailyReturns: '日收益率',
    dailyK: '日K',
  },
  tooltip: {
    open: '开',
    close: '收',
    low: '低',
    high: '高',
    change: '涨跌',
  },
  risk: {
    title: '风险分析',
    historicalVar: '历史 VaR (95%)',
    parametricVar: '参数 VaR (95%)',
    cvar: 'CVaR (95%)',
    currentDrawdown: '当前回撤',
    maxDrawdownLimit: '回撤上限',
    scenario: '场景',
    /** {unit} = currency.cny / currency.usd */
    varColumn: 'VaR（{unit}）',
    /** {value} = 基准 VaR 金额（fmtCurrency 输出） */
    stressTitle: '压力测试场景（基准 VaR：{value}）',
  },
}

export const en: Messages = {
  subtitle:
    'Pick a market, strategy and symbol, then run a backtest to see dynamic performance charts (Alpha/Beta, drawdown, daily returns)',
  running: 'Running backtest, please wait...',
  formRestored: 'Restored historical backtest parameters: {name} · {code}',
  strategiesLoadFailed: 'Failed to load strategies: ',
  runFailed: 'Backtest failed: ',
  done: 'Backtest completed',
  selectStrategyRequired: 'Please select a strategy',
  noTrades: 'No trades during the backtest period (the strategy produced no buy/sell signals)',
  avgCost: 'Avg Cost',
  market: {
    zh_a: 'A-shares',
    us: 'US Stocks',
  },
  currency: {
    cny: 'CNY',
    usd: 'USD',
  },
  benchmark: {
    csi300: 'CSI 300',
    sp500: 'S&P 500',
  },
  stockHint: {
    zh_a: '6-digit A-share code without an exchange prefix',
    us: 'US ticker symbols, e.g. AAPL, MSFT, BRK.B',
  },
  commissionHint: {
    zh_a: 'A-shares automatically include stamp duty (0.05% on sells) and transfer fee (0.001% both sides)',
    us: 'US stocks: commission only (both sides), no stamp duty or transfer fee',
  },
  form: {
    market: 'Market',
    strategy: 'Strategy',
    strategyPlaceholder: 'Select a strategy',
    stockCode: 'Stock Code',
    initialCapital: 'Initial Capital',
    commissionRate: 'Commission Rate',
    startDate: 'Start Date',
    endDate: 'End Date',
    run: 'Run Backtest',
  },
  section: {
    metrics: 'Performance Metrics',
    tradesKline: 'Backtest K-line · Trade Markers',
    equity: 'Cumulative Return',
    drawdown: 'Max Drawdown Curve',
    dailyReturns: 'Daily Return Distribution',
  },
  empty: {
    kline: 'K-line data failed to load, or the backtest range is invalid',
    result: 'Run a backtest to see dynamic performance charts here',
  },
  metric: {
    总收益率: 'Total Return',
    年化收益率: 'Annualized Return',
    夏普比率: 'Sharpe Ratio',
    最大回撤: 'Max Drawdown',
    胜率: 'Win Rate',
    盈亏比: 'Profit/Loss Ratio',
    信息比率: 'Information Ratio',
    年化波动率: 'Annualized Volatility',
    卡玛比率: 'Calmar Ratio',
    下行标准差: 'Downside Deviation',
    var95: 'VaR (95%)',
    cvar95: 'CVaR (95%)',
    交易次数: 'Trades',
    盈利交易数: 'Winning Trades',
    亏损交易数: 'Losing Trades',
  },
  series: {
    strategyNav: 'Strategy NAV',
    drawdown: 'Drawdown',
    dailyReturns: 'Daily Returns',
    dailyK: 'Daily K',
  },
  tooltip: {
    open: 'Open',
    close: 'Close',
    low: 'Low',
    high: 'High',
    change: 'Change',
  },
  risk: {
    title: 'Risk Analysis',
    historicalVar: 'Historical VaR (95%)',
    parametricVar: 'Parametric VaR (95%)',
    cvar: 'CVaR (95%)',
    currentDrawdown: 'Current Drawdown',
    maxDrawdownLimit: 'Max Drawdown Limit',
    scenario: 'Scenario',
    varColumn: 'VaR ({unit})',
    stressTitle: 'Stress Test Scenarios (Baseline VaR: {value})',
  },
}

File: src/utils/i18n_data/backtest.py
"""回测域文案目录：回测配置页（backtest_form.html）与回测结果页（backtest_result.html）。

「指标键标签」一节（templates 侧消费方式：

    {{ t('metric.' ~ k, k) }}   # k 为 performance_data/metrics 的中文键

``metric.*`` 的 key 段就是 ``backtest_manager._format_metrics_*`` 产出的中文指标名
（键名本身不动，SPA/types.ts 与 localStorage 存档依赖它）；未收录的键会由
``t(key, default)`` 的 default 兜底回中文原键，不会出现空标签。
其余键为表单标签/提示/按钮（``backtestForm.*``）、结果页分区标题（``backtestResult.*``）、
内联脚本文案（``js.backtest.*``）。zh 值必须等于改造前模板的现有中文文案。

**后端数据保持中文原样、不收录进目录**：``strategy_name``、``stock_code``、指标值
（``performance_data`` 的 value）、图表 PNG 本身（由可视化层按语言另出）。

``js.`` 前缀条目由 web_app 注入 ``window.__I18N__``，内联脚本用
``tt('js.backtest.xxx', '中文兜底')`` 读取（tt() 无参数支持，含占位符时用
``.replace('{name}', x)``，本文件暂无此类条目）。
"""

MESSAGES = {
    # ---- 指标键标签（backtest_result.html / SPA 指标卡片）----
    "metric.总收益率": {"zh": "总收益率", "en": "Total Return"},
    "metric.年化收益率": {"zh": "年化收益率", "en": "Annualized Return"},
    "metric.年化波动率": {"zh": "年化波动率", "en": "Annualized Volatility"},
    "metric.夏普比率": {"zh": "夏普比率", "en": "Sharpe Ratio"},
    "metric.最大回撤": {"zh": "最大回撤", "en": "Max Drawdown"},
    "metric.最大回撤开始时间": {"zh": "最大回撤开始时间", "en": "Drawdown Start"},
    "metric.最大回撤结束时间": {"zh": "最大回撤结束时间", "en": "Drawdown End"},
    "metric.卡玛比率": {"zh": "卡玛比率", "en": "Calmar Ratio"},
    "metric.胜率": {"zh": "胜率", "en": "Win Rate"},
    "metric.盈亏比": {"zh": "盈亏比", "en": "Profit/Loss Ratio"},
    "metric.信息比率": {"zh": "信息比率", "en": "Information Ratio"},
    "metric.下行标准差": {"zh": "下行标准差", "en": "Downside Deviation"},
    "metric.交易次数": {"zh": "交易次数", "en": "# Trades"},
    "metric.盈利交易数": {"zh": "盈利交易数", "en": "Winning Trades"},
    "metric.亏损交易数": {"zh": "亏损交易数", "en": "Losing Trades"},
    "metric.平均盈利": {"zh": "平均盈利", "en": "Avg Profit"},
    "metric.平均亏损": {"zh": "平均亏损", "en": "Avg Loss"},

    # ---- 回测配置页（backtest_form.html）----
    # 注意：键名后缀 ``Zh`` / ``Us`` 指**市场**（zh_a / us，A 股 / 美股），不是语言；
    # HTML 静态文案只有 A 股默认值，美股市值由内联脚本的 js.backtest.*Us 键替换。
    "backtestForm.heading": {"zh": "策略回测配置", "en": "Backtest Configuration"},
    "backtestForm.subtitle": {
        "zh": "选择市场、策略与标的，运行回测查看含 Alpha/Beta、最大回撤的详细绩效",
        "en": "Pick a market, strategy, and symbol, then run a backtest for a detailed performance report with Alpha/Beta and max drawdown",
    },
    "backtestForm.paramsTitle": {"zh": "回测参数", "en": "Backtest Parameters"},
    "backtestForm.market": {"zh": "市场", "en": "Market"},
    "backtestForm.marketZhA": {"zh": "A 股", "en": "A-shares"},
    "backtestForm.marketUs": {"zh": "美股", "en": "US Stocks"},
    "backtestForm.strategy": {"zh": "选择策略", "en": "Select Strategy"},
    "backtestForm.strategyPlaceholder": {"zh": "-- 请选择策略 --", "en": "-- Select a strategy --"},
    "backtestForm.stockCode": {"zh": "股票代码", "en": "Stock Code"},
    "backtestForm.stockCodePlaceholder": {"zh": "如 000001", "en": "e.g. 000001"},
    "backtestForm.stockHintZh": {
        "zh": "不带前缀的 6 位 A 股代码",
        "en": "6-digit A-share code without prefix",
    },
    "backtestForm.capitalZh": {"zh": "初始资金（元）", "en": "Initial Capital (CNY)"},
    "backtestForm.startDate": {"zh": "开始日期", "en": "Start Date"},
    "backtestForm.endDate": {"zh": "结束日期", "en": "End Date"},
    "backtestForm.commissionRate": {"zh": "佣金费率（双边）", "en": "Commission Rate (both sides)"},
    "backtestForm.commissionHintZh": {
        "zh": "A 股已自动叠加印花税（卖出 0.05%）与过户费（双边 0.001%）",
        "en": "A-share costs already include stamp duty (0.05% on sells) and transfer fee (0.001% both sides)",
    },
    "backtestForm.backHome": {"zh": "返回首页", "en": "Back to Home"},
    "backtestForm.run": {"zh": "运行回测", "en": "Run Backtest"},
    "backtestForm.instructions": {"zh": "使用说明", "en": "How to Use"},
    "backtestForm.howtoMarket": {"zh": "选择市场（A股 / 美股）", "en": "Choose a market (A-shares / US stocks)"},
    "backtestForm.howtoStrategy": {"zh": "选择已创建的策略", "en": "Choose an existing strategy"},
    "backtestForm.howtoCode": {
        "zh": "输入股票代码（A股如 000001，美股如 AAPL）",
        "en": "Enter a stock code (A-shares: 000001; US: AAPL)",
    },
    "backtestForm.howtoRun": {
        "zh": "运行后查看绩效指标与图表",
        "en": "Review performance metrics and charts after the run",
    },
    "backtestForm.benchmarkHint": {
        "zh": "A 股基准为沪深300；美股基准为标普500。美股仅佣金，无印花税/过户费。",
        "en": "The A-share benchmark is the CSI 300; the US benchmark is the S&P 500. US stocks incur commission only, with no stamp duty or transfer fee.",
    },

    # ---- 回测结果页（backtest_result.html）----
    # 标题形如「回测结果 · {{ strategy_name }}」，策略名为后端数据保持原样
    "backtestResult.heading": {"zh": "回测结果", "en": "Backtest Result"},
    "backtestResult.rerun": {"zh": "重新回测", "en": "Run Again"},
    "backtestResult.metricsTitle": {"zh": "绩效指标", "en": "Performance Metrics"},
    "backtestResult.equityTitle": {"zh": "累计收益曲线", "en": "Cumulative Return Curve"},
    "backtestResult.equityAlt": {"zh": "收益曲线", "en": "Equity curve"},
    "backtestResult.equityMissing": {
        "zh": "收益曲线图表未生成",
        "en": "The equity curve chart was not generated",
    },
    "backtestResult.drawdownTitle": {"zh": "最大回撤曲线", "en": "Max Drawdown Curve"},
    "backtestResult.drawdownAlt": {"zh": "回撤曲线", "en": "Drawdown curve"},
    "backtestResult.drawdownMissing": {
        "zh": "回撤曲线图表未生成",
        "en": "The drawdown chart was not generated",
    },
    "backtestResult.dashboardTitle": {"zh": "绩效仪表板", "en": "Performance Dashboard"},
    "backtestResult.dashboardAlt": {"zh": "绩效仪表板", "en": "Performance dashboard"},
    "backtestResult.dashboardMissing": {
        "zh": "绩效仪表板未生成",
        "en": "The performance dashboard was not generated",
    },

    # ---- 回测配置页内联脚本文案（js. 前缀，供 tt() 读取）----
    "js.backtest.codePlaceholderZh": {"zh": "如 000001", "en": "e.g. 000001"},
    "js.backtest.stockHintZh": {
        "zh": "不带前缀的 6 位 A 股代码",
        "en": "6-digit A-share code without prefix",
    },
    "js.backtest.capitalZh": {"zh": "初始资金（元）", "en": "Initial Capital (CNY)"},
    "js.backtest.commissionHintZh": {
        "zh": "A 股已自动叠加印花税（卖出 0.05%）与过户费（双边 0.001%）",
        "en": "A-share costs already include stamp duty (0.05% on sells) and transfer fee (0.001% both sides)",
    },
    "js.backtest.codePlaceholderUs": {"zh": "如 AAPL", "en": "e.g. AAPL"},
    "js.backtest.stockHintUs": {
        "zh": "美股字母代码，如 AAPL、MSFT、BRK.B",
        "en": "US ticker symbols, e.g. AAPL, MSFT, BRK.B",
    },
    "js.backtest.capitalUs": {"zh": "初始资金（美元）", "en": "Initial Capital (USD)"},
    "js.backtest.commissionHintUs": {
        "zh": "美股仅佣金（双边），无印花税/过户费",
        "en": "US stocks charge commission only (both sides); no stamp duty or transfer fee",
    },
    "js.backtest.running": {"zh": "正在运行回测，请稍候...", "en": "Running backtest, please wait..."},
}
